alacritty/src/renderer/rects.rs

```diff
@@ -190,3 +195,208 @@ impl RenderLines {
         }
     }
 }
+
+/// Shader sources for rect rendering program.
+pub static RECT_SHADER_F_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.f.glsl");
+pub static RECT_SHADER_V_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.v.glsl");
+static RECT_SHADER_F: &str = include_str!("../../res/rect.f.glsl");
+static RECT_SHADER_V: &str = include_str!("../../res/rect.v.glsl");
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
```

Comment:
Would it make sense to share this between the normal and the rect renderer? I know @kchibisov has mentioned something like this before, but it would be nice if things behaved uniformly.

It's also probably something that is useful in more places than just right here. Alternatively using `Rgb` from `alacritty_terminal` together with alpha would be another option.

Reply:
I think the problem is that you want a `repr(C)`, however alacritty_terminal structure isn't `repr(C)`, thus you can't really say that it'll work the way you want. Maybe rust will make it `repr(C)` if you start nesting, idk.

Reply:
I mean we're free to add whatever attributes we want. It's not like `repr(C)` on a struct with 4 elements is going to result in any weird inefficient behavior.

Though generally I'd say reusing `Rgb` from `alacritty_terminal` is probably the worst part of the suggestion, since we obviously need more than just `Rgb`.

Reply:
It's not clear to me what should I do here exactly.

This `Rgba` thing is maybe not great in greater context, but is locally-optimal, given clippy complaints and such (IMHO).

Reply:
It's mostly about consistency. Having it as fields in one struct and `Rgba` in another doesn't make much sense.

We should decide to do either one **or** the other. And once we've made that decision, I don't care what clippy says. We can ignore that warning if it doesn't make sense.

I personally don't care much about which one we choose, usually structs are nicer to work with but if raw types are easier for OpenGL interop then that's perfectly fine with me too.